pets/peer-finder/peer-finder.go

```diff
@@ -74,17 +75,42 @@ func main() {
 	if ns == "" {
 		ns = os.Getenv("POD_NAMESPACE")
 	}
-	if *svc == "" || ns == "" || (*onChange == "" && *onStart == "") {
-		log.Fatalf("Incomplete args, require -on-change and/or -on-start, -service and -ns or an env var for POD_NAMESPACE.")
-	}
-
 	hostname, err := os.Hostname()
 	if err != nil {
 		log.Fatalf("Failed to get hostname: %s", err)
 	}
+	var domainName string
+
+	// If domain is not provided, try to get it from resolv.conf
+	if (*domain == "") || (ns == "") {
+		resolvConfBytes, err := ioutil.ReadFile("/etc/resolv.conf")
+		resolvConf := string(resolvConfBytes)
+		if err != nil {
+			log.Fatal("Unable to read /etc/resolv.conf")
+		}
+		if ns != "" {
+			tempResolv := strings.Split(resolvConf, ("search " + ns + "."))
```

Comment:
> Probably should add something that says "This program relies on dnsPolicy set to ClusterFirst - would that work?

Yeah we should definitely clarify this in README. I think peer-finder works right currently in most of the cases --- the default value for dnsPolicy is `ClusterFirst` instead of `Default` :)